### PR TITLE
[10.0] mig stock_picking_package_preparation*

### PR DESCRIPTION
**Pull request by eLBati**

### CODE REVIEW

**Comment by eLBati** [comment]
Squashed
Please review

File: stock_picking_package_preparation/i18n/it.po [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+#. module: stock_picking_package_preparation
+#: model:ir.model.fields,field_description:stock_picking_package_preparation.field_stock_picking_package_preparation_id
+msgid "ID"
+msgstr "ID"
+
+#. module: stock_picking_package_preparation
+#: model:ir.ui.view,arch_db:stock_picking_package_preparation.stock_picking_package_preparation_form
+msgid "If there is no product but a source package, this means the source package was moved entirely.  If there is a product and a source package, the product was taken from the source package."
+msgstr "If there is no product but a source package, this means the source package was moved entirely.  If there is a product and a source package, the product was taken from the source package."
```
Comment (OpenCode) [inline stock_picking_package_preparation/i18n/it.po]: Da tradurre

File: stock_picking_package_preparation/i18n/it.po [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+#: model:ir.ui.view,arch_db:stock_picking_package_preparation.stock_picking_package_preparation_form
+msgid "Package Preparation"
+msgstr "Preparazione pacco"
+
+#. module: stock_picking_package_preparation
+#: model:ir.actions.act_window,name:stock_picking_package_preparation.action_stock_picking_package_preparation
+#: model:ir.ui.menu,name:stock_picking_package_preparation.menu_stock_picking_package_preparation
+#: model:ir.ui.view,arch_db:stock_picking_package_preparation.stock_picking_package_preparation_tree
+msgid "Package Preparations"
+msgstr "Package Preparations"
```
Comment (OpenCode) [inline stock_picking_package_preparation/i18n/it.po]: Da tradurre

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 3]
```diff
@@ -1,25 +1,11 @@
...
-##############################################################################
-
-from openerp import models, fields, api, exceptions, _
+# Copyright 2015 Guewen Baconnier
+# Copyright 2016 Lorenzo Battistini - Agile Business Group
+# Copyright 2016 Alessio Gerace - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+from odoo.exceptions import Warning
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Use from openerp.exceptions import Warning as UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 4]
```diff
@@ -107,67 +88,61 @@ def _default_company_id(self):
...
-        self.pack_operation_ids = self.mapped('picking_ids.pack_operation_ids')
+        for preparation in self:
+            preparation.pack_operation_ids = preparation.mapped(
+                'picking_ids.pack_operation_ids')
 
     @api.multi
     def action_done(self):
         if not self.mapped('package_id'):
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 5]
```diff
@@ -107,67 +88,61 @@ def _default_company_id(self):
...
-        self.picking_ids.do_transfer()
+        for picking in self.picking_ids:
+            picking.do_transfer()
         self.write({'state': 'done', 'date_done': fields.Datetime.now()})
 
     @api.multi
     def action_cancel(self):
         if any(prep.state == 'done' for prep in self):
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 6]
```diff
@@ -178,7 +153,7 @@ def action_cancel(self):
     @api.multi
     def action_draft(self):
         if any(prep.state != 'cancel' for prep in self):
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 7]
```diff
@@ -193,17 +168,16 @@ def action_put_in_pack(self):
     def _prepare_package(self):
         self.ensure_one()
         if not self.picking_ids:
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 8]
```diff
@@ -193,17 +168,16 @@ def action_put_in_pack(self):
...
                 _('No transfer selected for this preparation.')
             )
         location = self.mapped('picking_ids.location_dest_id')
         if len(location) > 1:
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

File: stock_picking_package_preparation/models/stock_picking_package_preparation.py [hunk 9]
```diff
@@ -212,11 +186,11 @@ def _prepare_package(self):
     def _generate_pack(self):
         self.ensure_one()
         pack_model = self.env['stock.quant.package']
-        move_model = self.env['stock.move']
+        moves = self.env['stock.move']
         operation_model = self.env['stock.pack.operation']
 
         if any(picking.state != 'assigned' for picking in self.picking_ids):
-            raise exceptions.Warning(
+            raise Warning(
```
Comment (OpenCode) [inline stock_picking_package_preparation/models/stock_picking_package_preparation.py]: Replace Warning with UserError

**Review by alessandrocamilli — APPROVED** [review]

**Review by OpenCode — COMMENTED** [review]
Technical review

**Review by OpenCode — APPROVED** [review]

**Review by tafaRU — APPROVED** [review]

**Pull request merged**